juju/juju
# [JUJU-2919] Merge decoupled api onto develop

Merges the `decouple-apiclient` branch into `develop`.

### QA Steps

Please confirm the source and target, i.e. double check that the right stuff is going into the right place.
